clash-lib/src/Clash/Backend/VHDL.hs

```diff
@@ -1463,6 +1476,14 @@ inst_' id_ scrut scrutTy es = fmap Just $
                                               <+> "else"
                                               <:> conds es'
 
+renderCompDecl :: (TextS.Text, [(TextS.Text, PortDirection, HWType)]) -> VHDLM Doc
+renderCompDecl (nm, ps0) =
+  "component" <+> pretty nm <+>
+    ("port" <> line <> indent 2 (tupledSemi ps <> semi))
+    <> line <> "end component" <> semi
+  where ps = traverse (\(t,pd,ty) -> pretty t <+> ":" <+> ppd pd <+> qualTyName ty) ps0
+        ppd = \case { In -> "in"; Out -> "out"}
```

Comment:
... and you don't need this special component-specific function, you can just use `inst_`